File: TextosJapones/src/main/java/org/jisho/textosJapones/controller/BaseController.java
package org.jisho.textosJapones.controller;

import javafx.scene.control.ProgressBar;
import javafx.scene.layout.AnchorPane;
import org.jisho.textosJapones.controller.novels.NovelsController;

public interface BaseController {
    AnchorPane getRoot();
    NovelsController getControllerPai();
    ProgressBar getBarraProgresso();
    void habilitar();
}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/controller/BaseController.kt
package br.com.fenix.processatexto.controller

import javafx.scene.control.ProgressBar
import javafx.scene.layout.AnchorPane
import javafx.scene.layout.StackPane


interface BaseController {
    val stackPane: StackPane
    val root: AnchorPane
    val barraProgresso: ProgressBar
    fun habilitar()
}